pkg/services/libraryelements/libraryelements_test.go

```diff
@@ -392,8 +365,22 @@ func validateAndUnMarshalArrayResponse(t *testing.T, resp response.Response) lib
 
 func scenarioWithPanel(t *testing.T, desc string, fn func(t *testing.T, sc scenarioContext)) {
 	t.Helper()
-	store := dbtest.NewFakeDB()
-	guardian.InitLegacyGuardian(setting.NewCfg(), store, &dashboards.FakeDashboardService{}, &teamtest.FakeService{})
+
+	sqlStore := db.InitTestDB(t)
+	ac := actest.FakeAccessControl{}
+	quotaService := quotatest.New(false, nil)
+	dashboardStore, err := database.ProvideDashboardStore(sqlStore, sqlStore.Cfg, featuremgmt.WithFeatures(), tagimpl.ProvideService(sqlStore, sqlStore.Cfg), quotaService)
+	require.NoError(t, err)
+	features := featuremgmt.WithFeatures()
+	folderPermissions := acmock.NewMockedPermissionsService()
+	dashboardPermissions := acmock.NewMockedPermissionsService()
+	folderStore := folderimpl.ProvideDashboardFolderStore(sqlStore)
+	dashboardService, err := dashboardservice.ProvideDashboardServiceImpl(
+		sqlStore.Cfg, dashboardStore, folderStore, nil,
+		features, folderPermissions, dashboardPermissions, ac,
+		foldertest.NewFakeService(),
+	)
+	guardian.InitAccessControlGuardian(sqlStore.Cfg, sqlStore, ac, folderPermissions, dashboardPermissions, dashboardService)
```

Comment:
this is the main change right? from legacy to AC

Reply:
This is one of them.

Roughly, the main changes are:
* enabling RBAC (removing `cfg.RBACEnabled = false`);
* switching to RBAC guardian from the legacy guardian (this change);
* setting RBAC permissions on folders (instead of relying on the old folder ACLs).

It was more work than I initially hoped it would be, as these tests have a lot of setup. I think ideally we could simplify the setup and reduce the dependencies, but that would require even more work now, and I don't have the capacity for it :(

Reply:
> setting RBAC permissions on folders (instead of relying on the old folder ACLs).

This is in part of the recent changes you made to nester folders

Reply:
Not really - we used to have ACLs for dashboards, folders, teams and data sources as part of the old access control system. This has been replaced by managed permissions with RBAC. So the way that permissions for folders are set is just completely different with RBAC than it was without.